### PR TITLE
[expo-google-app-auth] Make Google.logInAsync Behavior Consistent

### LINKED ISSUE #4655: [SDK 33] Google.logInAsync Unexpected behaviour on standalone
### Environment

Expo CLI 2.20.0 environment info:
-  System:
-- OS: Windows 10
- Binaries:
-- npm: 6.9.0 - C:\Program Files\nodejs\npm.CMD
- App Target:
-- Android, Standalone
- Testing Device:
-- Huawei P20 Lite, Android 8.

### Steps to Reproduce

The project uses `Google.logInAsync` to log in the users. It works fine on the development environment, but when the aplication is built and in standalone, at the time the user selects the account to log in, the browser window in which the selection whas made redirects to google's website, and the promise is never acomplished (only on the browser window's cancelation). 

### Expected Behavior

The `Google.logInAsync` promise is resolved/rejected

### Actual Behavior

On standalone when the user selects the account, the browser shows the loading animation for a few seconds, then redirects to google's website

### Reproducible Demo
I created a simple app with the example in [Expo's "Google" Docs](https://docs.expo.io/versions/latest/sdk/google/#__next) : https://snack.expo.io/BJvu0uDJB
and the problem persists (works as expected in expo client, crashes in standalone)

### PR DESCRIPTION
**Pull request by cruzach**
> Originally, this was done in [PR #4671](https://github.com/expo/expo/pull/4671), but after changes made in #4815, I just decided to close the original PR and make a new one to simplify

# Why

In its current state, `google.logInAsync` behaves differently in the Expo client and standalone (Android). Without providing a `redirectUrl` on Android standalone, after logging in it will navigate to google's homepage. Then, if you X out of that window, the login fails. However- in the Expo client, it signs in and navigates back to the app afterwards (successful sign in)
Seems to be caused by [this](https://github.com/expo/expo/blob/master/packages/expo/src/Google.ts#L151), which will assign a `redirectUrl` only if it's a standalone app (and the default it gives causes the issue stated above).

So basically:
- the `redirectUrl` prop will be ignored in **not-standalone** apps
- the default `redirectUrl` results in undesired behavior in standalone apps

Closes #4655 
# How

Just replaced the the standalone app check with a check for if a `redirectUrl` was provided. If it was provided, use that, if it wasn't, use `${AppAuth.OAuthRedirect}:/oauth2redirect/google` instead of `${reverseClientId}:/oauth2redirect/google` (not sure why the latter isn't working since [according to google](https://developers.google.com/identity/protocols/OAuth2InstalledApp#request-parameter-redirect_uri) they're both viable options)

### CODE REVIEW

**Comment by ide** [comment]
It seems like specifying `config.redirectUrl` could be a trap in the sense that some URLs will work in either the Expo client or in a standalone app, but not both. I think we need to document how to create a `redirectUrl` that works in both environments so that people don't end up with code that works in development but not in production, or vice versa.

Also fix this up:
```
🔍 Checking the expo-google-app-auth package ...
🏃‍♀️ Running `yarn clean` ...
🏃‍♀️ Running `yarn build` ...
The following build files need to be rebuilt and committed:
build/Google.js
build/Google.js.map
```

**Comment by ide** [comment]
Can you add a note to the changelog that this is a breaking change and how to upgrade? It's pretty subtle but everyone who specifies redirectUrl will need to inspect their code carefully.

**Comment by cruzach** [comment]
Forsure

For documenting creating a redirectUrl compatible for all the cases (expo client, standalone), you mean the following the instructions listed [here](https://docs.expo.io/versions/latest/workflow/linking/#linking-to-your-app)? Or is it more complicated than that

**Comment by ide** [comment]
I think namely saying that the URL usually should start with the value provided by `AppAuth.OAuthRedirect` so that it works in both the Expo client and in standalone apps. Then I think it'd be useful to lightly touch upon how `AppAuth.OAuthRedirect` differs between those two environments.

**Comment by cruzach** [comment]
I think it makes sense to state what the return value of `AppAuth.OAuthRedirect` in the AppAuth docs, and then link there from the google app auth docs

File: docs/pages/versions/unversioned/sdk/app-auth.md [hunk 1]
```diff
@@ -118,7 +118,7 @@ await AppAuth.revokeAsync(config, options);
 
 ## `AppAuth.OAuthRedirect`
 
-Redirect scheme used to assemble the `redirectUrl` prop.
+Redirect scheme used to assemble the `redirectUrl` prop. In standalone apps, this returns either the `android.package` (for Android) or `ios.bundleIdentifier` (for iOS) value from your `app.json`. However, for apps running in the Expo Client, `AppAuth.OAuthRedirect` returns `host.exp.exponent`.
```
Comment (ide) [inline docs/pages/versions/unversioned/sdk/app-auth.md]:
```suggestion
Redirect scheme used to assemble the `redirectUrl` prop. In standalone apps, this is either the `android.package` (for Android) or `ios.bundleIdentifier` (for iOS) value from your `app.json`. However, for apps running in the Expo Client, `AppAuth.OAuthRedirect` is `host.exp.exponent`. 
```

**Review by ide — APPROVED** [review]

**Comment by dtmuturi** [comment]
Hey @cruzach, I did a build earlier yesterday and the issue suddenly recurred! Google.js file in node_modules/expo/build/ has also changed breaking google login on my app. 

Any ideas how to fix this long term? Help Much appreciated.

**Comment by dtmuturi** [comment]
Ps, this only happens on standalone Android app using [Google](https://docs.expo.io/versions/latest/sdk/google/)

**Comment by vinagrito** [comment]
@cruzach The fix you made makes all the sense and after changing my code I can see that the config passed into the `logInAsync` function is the expected one - `com.example.my-example:/oauth2redirect/google`. 
And yet my app keeps redirecting to `google.com`. Any ideas/directions ?
